arches/app/media/js/views/graph/graph-designer/graph-tree.js

```diff
@@ -93,35 +94,24 @@ define([
 
         /**
          * Returns a boolean to indicate whether this node participates in descriptor function
-         * @param {object} node - a node in the tree 
+         * @param {object} node - a node in the tree
          */
         isFuncNode: function(node) {
-            var appFuncs = null, appFuncDesc = false, appFuncName = false;
-            if(this.appliedFunctions()) {
-                appFuncs = this.appliedFunctions();
-                for(var i = 0; i < appFuncs.length; i++) {
-                    if(appFuncs[i]['function_id'] == "60000000-0000-0000-0000-000000000001") {
-                        if(appFuncs[i]['config']['description']['nodegroup_id']) {
-                            appFuncDesc = appFuncs[i]['config']['description']['nodegroup_id'];
-                        }
-                        if(appFuncs[i]['config']['name']['nodegroup_id']) {
-                            appFuncName = appFuncs[i]['config']['name']['nodegroup_id'];
-                        }
-                        if(node['id'] === appFuncDesc || node['id'] === appFuncName) {
-                            return true;
-                        } else {
-                            if(node['children']) {
-                                node['children'].forEach( function(child) {
-                                    if(child['id'] === appFuncDesc || child['id'] === appFuncName) {
-                                        return true;
-                                    }
-                                }); 
-                            }
-                        }
-                    }
-                }
-            }
-            return false;
+            var primaryDescriptorNodes = {}, descriptorNode = null, pdFunction = this.primaryDescriptorFunction;
+
+            if(!this.primaryDescriptorFunction())
+                return null;
+
+            ['name', 'description'].forEach(function (descriptor) {
+                try {
+                    primaryDescriptorNodes[pdFunction()['config']['descriptor_types'][descriptor]['nodegroup_id']] = descriptor;
+                } catch (e) { } // Descriptor doesn't exist so ignore the exception
+            });
+
+            [node].concat(!!node['childNodes']() ? node['childNodes']() : [])
+                .find(nodeToCheck => !!(descriptorType = primaryDescriptorNodes[nodeToCheck['id']]));
```

Comment:
The `children` attribute didn't seem to exist. I've changed it to childNodes, which I think was the intent. I believe this was to highlight the parent node to make it easier to find the display node, however it can cause the tree root node to highlight which is a bit confusing. The node-forms.jsFuncNode has the same logic.

I'm wondering if the parents of the display nodes should be formatted slightly differently than the nodes that actually participate in the name/description function, or the parents not be highlighted at all.

Reply:
By display node, are you referring to the selected node? If that's the case aren't the parents of the selected node already styled differently (light grey as opposed to dark blue)?

Reply:
Ah, that's right. I think your change makes sense. 

As far as changing the highlighting of parents, I'm not sure what the right answer would be. Others would have to weigh in (maybe @dwuthrich, @robgaston, @apeters, @aj-he). The intent is to show editors where to find the cards that need to be completed in order to make the resource instance recognizable. For deeply nested branches the descriptor cards can be buried quite deep, so highlighting of the parents is intended to light the way. I think the assumption was made that users would understand that the lowest highlighted card is the target, but I can see how that might not be clear to people unfamiliar with the system.  

The drawback with highlighting parents differently is that we already have a color in the resource editor for cards that are provisional and another color for cards that have data, but are unsaved. If we add more colors, we're going to need a legend (maybe we are already at that point). On the other hand, if we were to remove the highlighting, the user would have to expand all of the cards in order to find the descriptor card(s). Would symbols rather than colors be better? I'm not sure.

In either case, we should probably create a different issue for this question.

Reply:
Created the issue for discussion here: #8260